query/src/pipelines/processors/sources/file_splitter.rs

```diff
@@ -0,0 +1,219 @@
+        reader: R,
+        input_format: Arc<dyn InputFormat>,
+        compress_algorithm: Option<CompressAlgorithm>,
+    ) -> FileSplitter {
+        let decoder = compress_algorithm.map(DecompressDecoder::new);
+        FileSplitter {
+            state: State::NeedData,
+            skipped_header: false,
+            reader: Box::new(reader),
+            input_buf: vec![],
+            decompress_buf: vec![],
+            decoder,
+            input_format: input_format.clone(),
+            format_state: input_format.create_state(),
+        }
+    }
+
+    pub fn state(&self) -> State {
+        self.state
+    }
+
+    fn split(&mut self, size: usize, output_splits: &mut VecDeque<Vec<u8>>) -> Result<()> {
+        //let data = &self.input_buf[..size];
+        if self.decoder.is_some() {
+            self.dec_and_split(size, output_splits)
+        } else {
+            Self::split_simple(
+                &self.input_buf[..size],
+                &self.input_format,
+                output_splits,
+                &mut self.format_state,
+                &mut self.skipped_header,
+            )
+        }
+    }
+
+    fn split_simple(
+        data: &[u8],
+        input_format: &Arc<dyn InputFormat>,
+        output_splits: &mut VecDeque<Vec<u8>>,
+        format_state: &mut Box<dyn InputState>,
+        skipped_header: &mut bool,
+    ) -> Result<()> {
+        let mut data_slice = data;
+
+        if !*skipped_header {
+            let len = data_slice.len();
+            let skip_size = input_format.skip_header(data_slice, format_state)?;
+            if skip_size < len {
+                *skipped_header = true;
+                *format_state = input_format.create_state();
+            }
+            data_slice = &data_slice[skip_size..];
+        }
+
+        while !data_slice.is_empty() {
+            let len = data_slice.len();
+            let read_size = input_format.read_buf(data_slice, format_state)?;
+
+            data_slice = &data_slice[read_size..];
+
+            if read_size < len {
+                let buf = input_format.take_buf(format_state);
+                output_splits.push_back(buf);
+            }
+        }
+        Ok(())
+    }
+
+    fn dec_and_split(&mut self, size: usize, output_splits: &mut VecDeque<Vec<u8>>) -> Result<()> {
+        let data = &self.input_buf[..size];
+        let decoder = self.decoder.as_mut().unwrap();
+        let mut amt = 0;
+
+        loop {
+            match decoder.state() {
+                DecompressState::Reading => {
+                    // If all data has been consumed, we should break with existing data directly.
+                    if amt == data.len() {
+                        break;
+                    }
+
+                    let read = decoder.fill(&data[amt..]);
+                    amt += read;
+                }
+                DecompressState::Decoding => {
+                    let written = decoder.decode(&mut self.decompress_buf).map_err(|e| {
```

Comment:
`decompress_buf`'s length seems to be `0`, so all `written` returned by  `DecompressDecoder` will be `0` too.

Reply:
Enable `RUST_LOG=opendal=debug` can display `compress_decoder` related logs, which will help debug.